lib/cacheOptimization.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+var gltfPrimitiveToCesiumGeometry = require('./gltfPrimitiveToCesiumGeometry');
+var cesiumGeometryToGltfPrimitive = require('./cesiumGeometryToGltfPrimitive');
+var createAccessorUsageTables = require('./createAccessorUsageTables');
+var Cesium = require('cesium');
+var GeometryPipeline = Cesium.GeometryPipeline;
+var defined = Cesium.defined;
+
+module.exports = cacheOptimization;
+
+// Helper method to map accessor collections from the usageTable to an independent primitive (if there is one)
+function createIndicesToAttributeDictionary(gltf) {
+    var dictionary = {};
+    var meshes = gltf.meshes;
+    for (var meshId in meshes) {
+        if (meshes.hasOwnProperty(meshId)) {
+            var mesh = meshes[meshId];
+            var primitives = mesh.primitives;
+            var primitivesLength = primitives.length;
+            for (var i = 0; i < primitivesLength; i++) {
+                var primitive = primitives[i];
+                var indicesId = primitive.indices;
+
+                var primitivesOfIndicesId;
+                if (!dictionary[indicesId]) {
+                    primitivesOfIndicesId = [];
+
+                } else {
+                    primitivesOfIndicesId = dictionary[indicesId];
+                }
+                primitivesOfIndicesId.push(primitive);
+
+                dictionary[indicesId] = primitivesOfIndicesId;
+            }
+        }
+    }
+    return dictionary;
+}
+
+function getIndependentPrimitive(dictionary, indicesId, attributeAccessors) {
+    var primitivesOfIndicesId = dictionary[indicesId];
+    for (var primitiveId in primitivesOfIndicesId) {
+        if (primitivesOfIndicesId.hasOwnProperty(primitiveId)) {
+            var primitive = primitivesOfIndicesId[primitiveId];
+            var accessors = primitive.attributes;
```

Comment:
Watch your naming here. Since the attributes are `SEMANTIC` : `accessorId` it would be better to call this `attributes` and iterate over a variable named `semantic` or `attributeSemantic`.